va/caa.go

```diff
@@ -196,7 +207,10 @@ func (va *ValidationAuthorityImpl) validateCAASet(caaSet *CAASet) (present, vali
 	//
 	// Our CAA identity must be found in the chosen checkSet.
 	for _, caa := range caaSet.Issue {
-		if extractIssuerDomain(caa) == va.issuerDomain {
+		caaIssuerDomain, caaParameters := extractIssuerDomainAndParameters(caa)
+		caaMethods, caaMethodsSet := caaParameters["validation-methods"]
+
+		if caaIssuerDomain == va.issuerDomain && (!features.Enabled(features.CAAValidationMethods) || !caaMethodsSet || challengeType == nil || containsMethod(caaMethods, *challengeType)) {
```

Comment:
There's a lot going on in this `if` statement. Could you break it up into a few smaller checks?